File: app/src/test/java/it/unical/mat/embasp/language/asp/ASPMapperTest.java
/**
 *
 */
package it.unical.mat.embasp.language.asp;

import java.lang.reflect.InvocationTargetException;

import org.junit.After;
import org.junit.AfterClass;
import org.junit.Assert;
import org.junit.Before;
import org.junit.BeforeClass;
import org.junit.Test;

import it.unical.mat.embasp.languages.IllegalAnnotationException;
import it.unical.mat.embasp.languages.ObjectNotValidException;
import it.unical.mat.embasp.languages.asp.ASPMapper;
import it.unical.mat.embasp.languages.asp.IllegalTermException;
import it.unical.mat.embasp.specializations.dlv.Cell;

public class ASPMapperTest {

	/**
	 * @throws java.lang.Exception
	 */
	@BeforeClass
	public static void setUpBeforeClass() throws Exception {
	}

	/**
	 * @throws java.lang.Exception
	 */
	@AfterClass
	public static void tearDownAfterClass() throws Exception {
	}

	/**
	 * @throws java.lang.Exception
	 */
	@Before
	public void setUp() throws Exception {
	}

	/**
	 * @throws java.lang.Exception
	 */
	@After
	public void tearDown() throws Exception {
	}

	@Test
	public void test() {

		final ASPMapper instance = ASPMapper.getInstance();

		try {

			instance.registerClass(Cell.class);

			final Object object = instance.getObject("cell(1,2,5)");

			Assert.assertTrue(object instanceof Cell);

			Assert.assertEquals(1, ((Cell) object).getRow());
			Assert.assertEquals(2, ((Cell) object).getColumn());
			Assert.assertEquals(5, ((Cell) object).getValue());

			Assert.assertEquals("cell(1,2,5)", instance.getString(object));

			instance.unregisterClass(Cell.class);
			
			final Object nullObject = instance.getObject("cell(1,2,5)");
			
			Assert.assertNull(nullObject);
			
			instance.registerClass(Arity0.class);
			
			final Object object1= instance.getObject("a");
			
			Assert.assertNotNull(object1);
			Assert.assertTrue(object1 instanceof Arity0);
			Assert.assertEquals("a", instance.getString(object1));

		} catch (ObjectNotValidException | IllegalAnnotationException | IllegalAccessException | IllegalArgumentException | InvocationTargetException
				| NoSuchMethodException | SecurityException | InstantiationException | IllegalTermException e) {
			Assert.fail(e.getMessage());
		}

	}

}

File: Python version/test/language/asp/asp_mapper_test.py
import unittest
from languages.asp.asp_mapper import ASPMapper
from test.language.asp.cell import Cell


class ASPMapperTest(unittest.TestCase):

    def runTest(self):

        instance = ASPMapper.get_instance()

        try:
            instance.register_class(Cell)

            obj = instance.get_object("cell(1,2,5)")

            self.assertTrue(isinstance(obj, Cell))

            self.assertEqual(1, obj.get_row())

            self.assertEqual(2, obj.get_column())

            self.assertEqual('5', obj.get_value().value)

            print(instance.get_string(obj))

            self.assertEqual("cell(1,2,5)", instance.get_string(obj))

            instance.unregister_class(Cell)

            noneObject = instance.get_object("cell(1,2,5)")

            self.assertIsNone(noneObject)

        except Exception as e:
            self.fail(str(e))


if __name__ == '__main__':
    unittest.main()
